### PR TITLE
docs(main): update hook doc

### PR DESCRIPTION
**Pull request by caohuilin**
### <samp>🤖 Generated by Copilot at f3a50aa</samp>

This pull request updates and adds documentation files for the `packages/document/main-doc/docs/en/apis/app/hooks` directory, which covers various features and configurations of the Modern.js web development framework. It improves the grammar, clarity, and accuracy of the existing files and deletes some obsolete files. It also adjusts the order of the files in the documentation sidebar.

### <samp>🤖 Generated by Copilot at f3a50aa</samp>

*  Add new documentation files for the `api` directory and its subdirectories, explaining the usage and conventions of the BFF feature in different modes and scenarios ([link](https://github.com/web-infra-dev/modern.js/pull/3775/files?diff=unified&w=0#diff-c8031710c00e33caa626bdb3b93af8dcd8b3143beb1b24fac6dbda01fe5bf3dfR1-R80), [link](https://github.com/web-infra-dev/modern.js/pull/3775/files?diff=unified&w=0#diff-18beb52030f56b5641a7d40b5bf2d179da665c35acc0a5fbc735c3ae3bcf6b87R1-R12), [link](https://github.com/web-infra-dev/modern.js/pull/3775/files?diff=unified&w=0#diff-2a9557530b90072d2f289b61b9ef25aba45ef4460b455bdc243f102535122770R1-R57))
*  Adjust the order of the documentation files in the sidebar by modifying the `sidebar_position` of the `test.mdx` file ([link](https://github.com/web-infra-dev/modern.js/pull/3775/files?diff=unified&w=0#diff-82ae318310eb1a667d1126728b057ede65bbea331fd157034ad094d20f380366L3-R3))
*  Rewrite the content of the existing documentation files for the `html`, `mock`, `public`, `storybook`, and `upload` directories, improving the grammar and clarity of the explanations ([link](https://github.com/web-infra-dev/modern.js/pull/3775/files?diff=unified&w=0#diff-38dc3d7dfcd25fd0bd50327e5a46e15d0aff009b62d393430808f39aa128a22bL7-R9), [link](https://github.com/web-infra-dev/modern.js/pull/3775/files?diff=unified&w=0#diff-6e081b7732d40b516a18296473f7d14bd027e46c1cb7df3889ea1c667bd572faL7-R7), [link](https://github.com/web-infra-dev/modern.js/pull/3775/files?diff=unified&w=0#diff-cd4eac45c04441b31c12764c64e6eeafe2b71c96270f10c144b91afe578b4d97L7-R30), [link](https://github.com/web-infra-dev/modern.js/pull/3775/files?diff=unified&w=0#diff-02f3a80980d5499681c6bb41f362a46a743c29a6a4e2c4b7823d3097c3748515L7-R12), [link](https://github.com/web-infra-dev/modern.js/pull/3775/files?diff=unified&w=0#diff-1e7a8db5fffc4c9ef27886eb2c85f7f8cb518d845b3c827a6b5229215e3e719fL7-R29))
*  Modify the `title` and rewrite the content of the existing documentation file for the `modern.config.[tj]s` file, improving the grammar and clarity of the explanations ([link](https://github.com/web-infra-dev/modern.js/pull/3775/files?diff=unified&w=0#diff-95e21269136cdef4c1f750e72353c97f165d692bbdb907b0165105b141505113L2-R9))
*  Rewrite the content of the existing documentation files for the `index.[tj]s` and `test.[tj]s` files under the `server` directory, improving the grammar and clarity of the explanations ([link](https://github.com/web-infra-dev/modern.js/pull/3775/files?diff=unified&w=0#diff-5b4101fd782dcc689ce108a91b8f1fa7d6c8b8a39fb6bba08966fd9977f0c564L8-R10), [link](https://github.com/web-infra-dev/modern.js/pull/3775/files?diff=unified&w=0#diff-fa12db6ba7a862c3563447e4556410fff4a95146c2652b563171436272fb3469L5-R13))
*  Rewrite the content of the existing documentation file for the `shared` directory, improving the grammar and clarity of the explanations ([link](https://github.com/web-infra-dev/modern.js/pull/3775/files?diff=unified&w=0#diff-e26884009f64a035e2a4805c9b049935d81f559bc48c625e40636f44261d95c3L7-R7))
*  Modify the `title` of the existing documentation file for the `icon` directory, matching the convention of using directory names as titles ([link](https://github.com/web-infra-dev/modern.js/pull/3775/files?diff=unified&w=0#diff-b3010b3ed8e72921e5e119bf2b3215c230d4c231d7d6e50dd5620c5c1ba8f3d4L2-R10))
*  Rewrite some sentences in the existing documentation file for the `icon` directory, improving the grammar and clarity of the explanations ([link](https://github.com/web-infra-dev/modern.js/pull/3775/files?diff=unified&w=0#diff-b3010b3ed8e72921e5e119bf2b3215c230d4c231d7d6e50dd5620c5c1ba8f3d4L17-R17), [link](https://github.com/web-infra-dev/modern.js/pull/3775/files?diff=unified&w=0#diff-b3010b3ed8e72921e5e119bf2b3215c230d4c231d7d6e50dd5620c5c1ba8f3d4L25-R35), [link](https://github.com/web-infra-dev/modern.js/pull/3775/files?diff=unified&w=0#diff-b3010b3ed8e72921e5e119bf2b3215c230d4c231d7d6e50dd5620c5c1ba8f3d4L42-R42), [link](https://github.com/web-infra-dev/modern.js/pull/3775/files?diff=unified&w=0#diff-b3010b3ed8e72921e5e119bf2b3215c230d4c231d7d6e50dd5620c5c1ba8f3d4L50-R56))
*  Rewrite some sentences in the existing documentation file for the `upload` directory, improving the grammar and clarity of the explanations ([link](https://github.com/web-infra-dev/modern.js/pull/3775/files?diff=unified&w=0#diff-1e7a8db5fffc4c9ef27886eb2c85f7f8cb518d845b3c827a6b5229215e3e719fL39-R39), [link](https://github.com/web-infra-dev/modern.js/pull/3775/files?diff=unified&w=0#diff-1e7a8db5fffc4c9ef27886eb2c85f7f8cb518d845b3c827a6b5229215e3e719fL45-R45), [link](https://github.com/web-infra-dev/modern.js/pull/3775/files?diff=unified&w=0#diff-1e7a8db5fffc4c9ef27886eb2c85f7f8cb518d845b3c827a6b5229215e3e719fL52-R52))
*  Delete some unused documentation files for the `api` directory and its subdirectories, removing redundant or outdated information ([link](https://github.com/web-infra-dev/modern.js/pull/3775/files?diff=unified&w=0#diff-81dc5bcf5674b84ee1be055cdb9749420e9a31a55f5b22c1360cc07f62872e0d), [link](https://github.com/web-infra-dev/modern.js/pull/3775/files?diff=unified&w=0#diff-0664d66a2a6fa02c169cd63b696042e73595e2505456e4d140ce8418a988fd91), [link](https://github.com/web-infra-dev/modern.js/pull/3775/files?diff=unified&w=0#diff-161e4b4e9ac139876c36a37deb081a08aa8a96a0ac59723adfb24f1c1e131a59), [link](https://github.com/web-infra-dev/modern.js/pull/3775/files?diff=unified&w=0#diff-8310c1e5fb9398e6cbf4fbac77a3bcf678a3008ebe55fea7027fea2ded65b970), [link](https://github.com/web-infra-dev/modern.js/pull/3775/files?diff=unified&w=0#diff-a03cd84d2570d0820b494bbfac1e8419b258717bd86f89e6b98ceeb403cbd705), [link](https://github.com/web-infra-dev/modern.js/pull/3775/files?diff=unified&w=0#diff-6dfdd314fa7b459c3902bc9c487ffd3dc53adcee971890d364e7538923401183))

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/document/main-doc/docs/en/apis/app/hooks/api/app.mdx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
+title: _app.[tj]s
+sidebar_position: 2
+---
+# _app.[tj]s
```
Comment (yimingjfe) [inline packages/document/main-doc/docs/en/apis/app/hooks/api/app.mdx:5]: There is no function mode anymore.

File: packages/document/main-doc/docs/en/apis/app/hooks/api/lambda.mdx [hunk 2]
```diff
@@ -0,0 +1,57 @@
+---
+title: lambda/*.[tj]s
+sidebar_position: 3
+---
+# lambda/*.[tj]s
+
+Files that declare API routes under the [BFF Framework Mode](/guides/advanced-features/bff/type.html#framework-mode). Except for [convention files](/apis/app/hooks/api/api#allow-list), files under the `lambda/` directory will be registered as the routes.
```
Comment (yimingjfe) [inline packages/document/main-doc/docs/en/apis/app/hooks/api/lambda.mdx:7]: Is "In the deirectory" more appropriate?

File: packages/document/main-doc/docs/en/apis/app/hooks/api/api.mdx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+- Files whose names start with `_`. For example: `_utils.ts`.
+- All files under folders whose names start with `_`. For example: `_utils/index.ts`, `_utils/cp.ts`.
+- Test files. For example: `foo.test.ts`.
+- TypeScript type files. For example: `hello.d.ts`.
+- Files under `node_modules`.
+
+## Function Definition
+
+In addition to the routing rules above, there are also corresponding conventions for function definitions and exports in the code.
```
Comment (yimingjfe) [inline packages/document/main-doc/docs/en/apis/app/hooks/api/api.mdx:54]: "corresponding" is unnecessary.

File: packages/document/main-doc/docs/en/apis/app/hooks/src/routes.mdx [hunk 4]
```diff
@@ -50,35 +50,15 @@ If the directory name of the route file is named with `[]`, the generated route
...
-In the component, you can get the corresponding parameters by [useParams](/apis/app/runtime/router/router#useparams).
+In the component, you can use [useParams](/apis/app/runtime/router/router#useparams) to obtain the corresponding named parameter.
 
-In the loader, params will be used as input to [loader](/guides/basic-features/data-fetch#loader-function), and the corresponding parameters can be retrieved through the property `params`.
+When using the [loader](/guides/basic-features/data-fetch#the-loader-function) function to obtain data, `params` will be passed as an input parameter to the `loader` function, and the corresponding parameter can be obtained through the attribute of `params`.
 
-## Dynamic Optional Routes
+## Layout Component
 
-By using file directories named with `[$]`, the generated routes will be treated as dynamic routes. For example, the following file directory:
```
Comment (yimingjfe) [inline packages/document/main-doc/docs/en/apis/app/hooks/src/routes.mdx:61]: Why was this part deleted?

Comment (caohuilin) [reply packages/document/main-doc/docs/en/apis/app/hooks/src/routes.mdx:61]: The zh documentation no longer has this section, do you want me to add it back?

File: packages/document/main-doc/docs/en/apis/app/hooks/api/api.mdx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+### Default Routing
+
+The routing system will map files named `index` to the previous directory.
+
+- `api/index.ts` -> `$BASENAME/`
+- `api/user/index.ts` -> `$BASENAME/user`
+
+### Nested Routing
+
+The routing system also supports parsing nested files. If you create a nested folder structure, the files will still automatically resolve routes in the same way.
```
[the remark this answers is not in the document]
Comment (caohuilin) [reply packages/document/main-doc/docs/en/apis/app/hooks/api/api.mdx:27]: "the files" feel more appropriate, specifically referring to the files that use nested routing.

File: packages/document/main-doc/docs/en/apis/app/hooks/src/routes.mdx (hunk 4 above)
Comment (yimingjfe) [reply packages/document/main-doc/docs/en/apis/app/hooks/src/routes.mdx:61]: The zh documentation still has this section，[动态可选路由](https://github.com/web-infra-dev/modern.js/blob/HEAD/packages/document/main-doc/docs/zh/guides/basic-features/routes.mdx#%E5%8A%A8%E6%80%81%E5%8F%AF%E9%80%89%E8%B7%AF%E7%94%B1)

Comment (caohuilin) [reply packages/document/main-doc/docs/en/apis/app/hooks/src/routes.mdx:61]: This is  guide chapter，hooks chapter is not has this content

**Review by yimingjfe — APPROVED** [review]

File: packages/document/main-doc/docs/en/apis/app/hooks/src/routes.mdx (hunk 4 above)
Comment (yimingjfe) [reply packages/document/main-doc/docs/en/apis/app/hooks/src/routes.mdx:61]: I get it, done.

**Pull request merged**